crates/connectors/src/connector_s.rs

```diff
@@ -106,18 +109,27 @@ impl ConnectorStake {
     /// <stake_preimage> OP_EQUALVERIFY <ΔS> OP_CHECKSEQUENCEVERIFY
     /// ```
     pub fn generate_script(&self) -> ScriptBuf {
-        ScriptBuf::builder()
+        let mut locking_script = ScriptBuf::builder()
             .push_slice(self.operator_pubkey.serialize())
             .push_opcode(OP_CHECKSIGVERIFY)
             .push_opcode(OP_SIZE)
             .push_int(0x20)
             .push_opcode(OP_EQUALVERIFY)
             .push_opcode(OP_SHA256)
             .push_slice(self.stake_hash.to_byte_array())
-            .push_opcode(OP_EQUALVERIFY)
-            .push_sequence(self.delta.into())
-            .push_opcode(OP_CSV)
-            .into_script()
+            .push_opcode(OP_EQUALVERIFY);
+
+        // handle `0`-locktime differently as pushing `0` sequence means no element is pushed which
+        // results in the stack being empty when it is executed when spending.
+        if self.delta != LockTime::ZERO {
```

Comment:
You should not import the `LockTime` from the `relative` module in rust-bitcoin.
This may bite you in the future because we have two different `LockTime`s:

- `absolute::LockTime`
- `relative::LockTime`

For best practices, I always adhere to using the module-level qualification.
In other words instead of `LockTime` use `relative::LockTime`.

Reply:
And then you can undo the import changes in the top of the file.